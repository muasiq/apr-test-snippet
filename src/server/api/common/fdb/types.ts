import { z } from 'zod';
import { dispensableDrugSchema, interactionsSchema } from './schema';

export type DispensableDrug = z.infer<typeof dispensableDrugSchema>;

export type Interactions = z.infer<typeof interactionsSchema>;

export type FDBSearchType = 'Contains' | 'StartsWith' | 'TextEqual' | 'All' | 'Exhaustive';
