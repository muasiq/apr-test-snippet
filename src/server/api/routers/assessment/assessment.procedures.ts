import { createTRPCRouter, permissionProtectedProcedure } from '~/server/api/trpc';
import { UserActionPermissions } from '../../../../common/utils/permissions';
import * as assessmentInputs from './assessment.inputs';
import * as assessmentService from './assessment.service';

export const assessmentRouter = createTRPCRouter({
	assessmentCheck: permissionProtectedProcedure(UserActionPermissions.ANSWER_ASSESSMENT_QUESTION)
		.input(assessmentInputs.assessmentCheckSchema)
		.mutation(({ input, ctx }) => {
			return assessmentService.assessmentCheck(input, ctx);
		}),
	unicornButton: permissionProtectedProcedure(UserActionPermissions.AUTOFILL_ALL_DOCUMENTATION_ITEMS)
		.input(assessmentInputs.unicornButton)
		.mutation(({ input, ctx }) => {
			return assessmentService.unicornButtonMutation(input, ctx);
		}),
	verifyMedicationInteractionsAssessment: permissionProtectedProcedure(
		UserActionPermissions.VERIFY_MEDICATION_INTERACTIONS,
	)
		.input(assessmentInputs.verifyMedicationInteractionsAssessmentSchema)
		.mutation(({ input, ctx }) => {
			return assessmentService.verifyMedicationInteractionsAssessment(input, ctx);
		}),
	getAllAssessmentQuestionsForPatient: permissionProtectedProcedure(UserActionPermissions.VIEW_PATIENT_PROFILE)
		.input(assessmentInputs.getAllAssessmentQuestionsForPatientSchema)
		.query(({ input, ctx }) => {
			return assessmentService.getAllAssessmentQuestionsForPatient(input, ctx);
		}),

	getAssessmentQuestionForPatient: permissionProtectedProcedure(UserActionPermissions.VIEW_PATIENT_PROFILE)
		.input(assessmentInputs.getAssessmentQuestionForPatient)
		.query(({ input, ctx }) => {
			return assessmentService.getAssessmentQuestionForPatient(input, ctx);
		}),

	generateAssessmentAnswer: permissionProtectedProcedure(UserActionPermissions.GENERATE_SUGGESTION)
		.input(assessmentInputs.generateAssessmentAnswer)
		.mutation(({ input, ctx }) => {
			return assessmentService.generateAssessmentAnswer(input, ctx);
		}),
	generateAllAssessmentSuggestionsForPatient: permissionProtectedProcedure(UserActionPermissions.QA_TAB_VISIBLE)
		.input(assessmentInputs.generateAllAssessmentSuggestionsForPatient)
		.mutation(({ input, ctx }) => {
			return assessmentService.generateAllAssessmentSuggestionsForPatient(input, ctx);
		}),
});
