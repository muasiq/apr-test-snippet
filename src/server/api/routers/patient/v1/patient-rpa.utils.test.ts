import { expect, it } from '@jest/globals';
import { AssessmentAnswer, AssessmentAnswerIntegrationStatus } from '@prisma/client';
import { JsonValue } from '@prisma/client/runtime/library';
import { CustomAssessmentNumber } from '~/assessments/apricot/custom';
import { DataRendererOptions } from '~/assessments/apricot/data-renderers/options';
import { getLocalConfiguration } from '~/assessments/configurations';
import { backOfficeCompletedConfig } from '~/assessments/home-care-home-base/orgs/accentra/backOfficeCompletedConfig';
import { AssessmentQuestionSources, AssessmentQuestionType, QuestionSource } from '~/assessments/types';
import {
	expectedMedicationRpaResponse,
	sampleConfig,
	testMedicationCheckedResponse,
} from '~/common/testUtils/generateFakeAnswers';
import { samplePatientWithJoins } from '~/common/testUtils/seedPatients';
import { CheckedResponseUnion } from '~/server/api/routers/assessment/assessment.inputs';
import {
	RpaAssessmentAnswerWithQuestionText,
	generateAnswer,
	getAnswersForQuestionSource,
	getGeneratedBlankAnswers,
	getOriginalResponseText,
	rollupQuestionsMarkedForRollup,
} from './patient-rpa.utils';

const sampleQuestions: RpaAssessmentAnswerWithQuestionText[] = [
	generateAnswer('AC2022', 'Test 1', 'Yes', AssessmentQuestionType.FreeForm),
	generateAnswer('AC2024', 'Test 2', 'No', AssessmentQuestionType.FreeForm),
	generateAnswer('no-roll-up', 'Test 3', 'Test', AssessmentQuestionType.FreeForm),
];

describe('Patient RPA Utils', () => {
	describe('rollupQuestionsMarkedForRollup', () => {
		it('Should rollup any items marked for rollup', () => {
			const result = rollupQuestionsMarkedForRollup(backOfficeCompletedConfig, sampleQuestions);

			expect(result).toEqual([
				{
					assessmentNumber: 'no-roll-up',
					checkedResponse: {
						choice: 'Test',
					},
					integrationMessage: null,
					integrationStatus: 'Pending',
					mappedType: 'FreeForm',
					questionText: 'Test 3',
				},
				{
					assessmentNumber: 'start_of_care_narrative_note',
					checkedResponse: {
						choice: `Test 1: Yes\nTest 2: No\n`,
					},
					integrationMessage: null,
					integrationStatus: 'Pending',
					mappedType: 'FreeForm',
					questionText: 'Start of Care Narrative Note',
				},
			]);
		});
	});
});

describe('getAllGeneratedAnswers', () => {
	it('Get all autogenerated answers', () => {
		const allFields = getGeneratedBlankAnswers();
		expect(allFields).toMatchSnapshot();
	});
});

describe('getOriginalResponseText', () => {
	const config = getLocalConfiguration('Accentra');
	const physicalAssessmentQuestion = config.allQuestions.find((q) => q.id === 'PA-97a6bd')!;
	const testAnswer: Partial<AssessmentAnswer> = {
		checkedResponse: { choice: ['Patient Name', 'Date of Birth'] },
	};

	it('Returns original text in checkedResponse.choice', () => {
		expect(physicalAssessmentQuestion).toBeDefined();
		const response = getOriginalResponseText(
			testAnswer.checkedResponse as CheckedResponseUnion,
			physicalAssessmentQuestion,
		);
		expect(response).toEqual({ choice: ['PATIENT NAME', 'DATE OF BIRTH'] });
	});
});

describe('getAnswersForQuestionSource', () => {
	it('Should return answers for a given question source', () => {
		const sampleAssessments: Partial<AssessmentAnswer>[] = [
			{
				assessmentNumber: 'A1005',
				checkedResponse: { choice: ['Yes, Mexican, Mexican American, Chicano/a'] },
				integrationMessage: null,
				integrationStatus: 'Pending',
			},
		];
		const questionSource = { id: 'A1005', source: AssessmentQuestionSources.Oasis } as QuestionSource;
		const answers = getAnswersForQuestionSource(
			questionSource,
			samplePatientWithJoins,
			sampleAssessments as AssessmentAnswer[],
			sampleConfig,
		);
		expect(answers).toEqual({
			assessmentNumber: 'A1005',
			questionText: '(A1005) ARE YOU OF HISPANIC, LATINO/A, OR SPANISH ORIGIN? CHECK ALL THAT APPLY.',
			mappedType: 'SelectAllThatApply',
			checkedResponse: { choice: ['B. YES, MEXICAN, MEXICAN AMERICAN, CHICANO/A'] },
			integrationMessage: null,
			integrationStatus: 'Pending',
		});
	});
	it('Should return auto rendered answers', () => {
		const questionSource = { patientDataResolver: DataRendererOptions.DRIVE_TIME } as QuestionSource;
		const answers = getAnswersForQuestionSource(questionSource, samplePatientWithJoins, [], sampleConfig);
		expect(answers).toEqual([
			{
				assessmentNumber: 'CUS09',
				questionText: 'Drive time',
				mappedType: 'FreeForm',
				checkedResponse: { choice: 25 },
				integrationMessage: null,
				integrationStatus: 'Pending',
			},
		]);
	});

	it('Should return medication answers', () => {
		const testAssessment: Partial<AssessmentAnswer> = {
			assessmentNumber: CustomAssessmentNumber.MEDICATIONS,
			checkedResponse: { choice: { medications: [testMedicationCheckedResponse] as JsonValue } },
			integrationMessage: null,
			integrationStatus: 'Pending',
			status: 'Completed',
		};
		const questionSource = {
			id: CustomAssessmentNumber.MEDICATIONS,
			source: AssessmentQuestionSources.Custom,
		} as QuestionSource;
		const answers = getAnswersForQuestionSource(
			questionSource,
			samplePatientWithJoins,
			[testAssessment as AssessmentAnswer],
			sampleConfig,
		);
		expect(answers).toEqual({
			assessmentNumber: CustomAssessmentNumber.MEDICATIONS,
			checkedResponse: { choice: { medications: [expectedMedicationRpaResponse] } },
			integrationMessage: null,
			integrationStatus: 'Pending',
			mappedType: 'Medication',
			questionText: 'Medication List',
		});
	});

	it('should return original text for responses', () => {
		const questionSource = {
			id: 'O0110_A',
			source: AssessmentQuestionSources.Oasis,
		} as QuestionSource;
		const answer = {
			mappedType: 'SelectAllThatApply',
			integrationMessage: null,
			integrationStatus: 'Pending',
			assessmentNumber: 'O0110_A',
			checkedResponse: {
				choice: [
					'Oxygen therapy',
					'IV medications',
					'IV Access',
					'Tracheostomy care',
					'Chemotherapy',
					'Radiation',
					'Suctioning',
					'Invasive mechanical ventilator (ventilator or respirator)',
					'Non-invasive mechanical ventilator',
					'Transfusions',
					'Dialysis',
				],
			},
		} as Partial<AssessmentAnswer>;
		const answers = getAnswersForQuestionSource(
			questionSource,
			samplePatientWithJoins,
			[answer] as AssessmentAnswer[],
			sampleConfig,
		);
		expect(answers).toEqual({
			assessmentNumber: 'O0110_A',
			checkedResponse: {
				choice: [
					'C1. OXYGEN',
					'H1. IV MEDICATIONS',
					'O1. IV ACCESS',
					'E1. TRACHEOSTOMY CARE',
					'A1. CHEMOTHERAPY',
					'B1. RADIATION',
					'D1. SUCTIONING',
					'F1. INVASIVE MECHANICAL VENTILATOR (VENTILATOR OR RESPIRATOR)',
					'G1. NON-INVASIVE MECHANICAL VENTILATOR',
					'I1. TRANSFUSIONS',
					'J1. DIALYSIS',
				],
			},
			integrationMessage: null,
			integrationStatus: 'Pending',
			mappedType: 'SelectAllThatApply',
			questionText:
				'(O0110A) SPECIAL TREATMENTS, PROCEDURES, AND PROGRAMS: CHECK ALL OF THE FOLLOWING TREATMENTS, PROCEDURES, AND PROGRAMS THAT APPLY ON ADMISSION.',
		});
	});

	it('should strip TYPE annotations from originalText fields', () => {
		const questionSource = {
			id: 'PA-f0dc66',
			source: AssessmentQuestionSources.PhysicalAssessment,
		} as QuestionSource;
		const answers = getAnswersForQuestionSource(
			questionSource,
			samplePatientWithJoins,
			[] as AssessmentAnswer[],
			sampleConfig,
		);
		expect(answers).toEqual({
			assessmentNumber: 'PA-f0dc66',
			checkedResponse: {
				choice: null,
			},
			integrationMessage: null,
			integrationStatus: 'Completed',
			mappedType: 'SelectOne',
			questionText: 'WERE NOSE AND SINUSES ASSESSED?',
		});
	});

	it('should strip TYPE annotations from originalText fields for soc standard assessment', () => {
		const questionSource = {
			id: 'EV01',
			source: AssessmentQuestionSources.SOCStandardAssessment,
		} as QuestionSource;
		const answers = getAnswersForQuestionSource(
			questionSource,
			samplePatientWithJoins,
			[] as AssessmentAnswer[],
			sampleConfig,
		);
		expect(answers).toEqual({
			assessmentNumber: 'EV01',
			checkedResponse: {
				choice: null,
			},
			integrationMessage: null,
			integrationStatus: 'Completed',
			mappedType: 'SelectOne',
			questionText: 'DOES THE PATIENT HAVE MEDICARE?',
		});
	});
});

describe('getAnswersForQuestionSource', () => {
	it('should return answers for a given question source', () => {
		const sampleAssessments: Partial<AssessmentAnswer>[] = [
			{
				assessmentNumber: 'PA-97a6bd',
				checkedResponse: { choice: 'mock choice' },
				integrationMessage: null,
				integrationStatus: 'Completed',
				patientId: samplePatientWithJoins.id,
				status: 'Completed',
			},
		];
		const questionSource = {
			id: 'PA-97a6bd',
			source: AssessmentQuestionSources.PhysicalAssessment,
		} as QuestionSource;

		const sampleConfig = getLocalConfiguration('Accentra');

		const answers = getAnswersForQuestionSource(
			questionSource,
			samplePatientWithJoins,
			sampleAssessments as AssessmentAnswer[],
			sampleConfig,
		);

		// TODO: return value of getAnswersForQuestionSource could be an array or an object; could this be consolidated?
		if (typeof answers === 'object' && !Array.isArray(answers)) {
			expect(answers.questionText).toBe('INDICATE TWO PATIENT IDENTIFIERS USED TO IDENTIFY THE PATIENT:');
			expect(answers.integrationStatus).toBe(AssessmentAnswerIntegrationStatus.Completed);
		} else if (Array.isArray(answers) && answers.length > 0) {
			expect(answers[0]?.questionText).toBe('INDICATE TWO PATIENT IDENTIFIERS USED TO IDENTIFY THE PATIENT');
		}
	});

	it('should return auto rendered answers', () => {
		const questionSource = { patientDataResolver: DataRendererOptions.DRIVE_TIME } as QuestionSource;
		const answers = getAnswersForQuestionSource(questionSource, samplePatientWithJoins, [], sampleConfig);
		expect(answers).toEqual([
			{
				assessmentNumber: 'CUS09',
				questionText: 'Drive time',
				mappedType: 'FreeForm',
				checkedResponse: { choice: 25 },
				integrationMessage: null,
				integrationStatus: 'Pending',
			},
		]);
	});
});
