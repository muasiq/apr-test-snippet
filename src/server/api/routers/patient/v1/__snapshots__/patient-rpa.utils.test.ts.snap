// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllGeneratedAnswers Get all autogenerated answers 1`] = `
[
  {
    "assessmentNumber": "M0064",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "SSN",
  },
  {
    "assessmentNumber": "CUS11",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Branch",
  },
  {
    "assessmentNumber": "M0066",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "Date",
    "questionText": "Birth Date",
  },
  {
    "assessmentNumber": "CUS09",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Drive time",
  },
  {
    "assessmentNumber": "M0069",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "Gender",
  },
  {
    "assessmentNumber": "CUS10",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Mileage",
  },
  {
    "assessmentNumber": "M0040_A",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "First Name",
  },
  {
    "assessmentNumber": "M0040_B",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Middle Initial",
  },
  {
    "assessmentNumber": "M0040_C",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Last Name",
  },
  {
    "assessmentNumber": "M0040_D",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Suffix",
  },
  {
    "assessmentNumber": "M0050",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "Patient State of Residence",
  },
  {
    "assessmentNumber": "M0060",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Zip Code",
  },
  {
    "assessmentNumber": "CUS26",
    "checkedResponse": {
      "choice": {
        "address1": undefined,
        "address2": undefined,
        "city": undefined,
        "county": undefined,
        "notes": undefined,
        "serviceLocationType": undefined,
        "state": undefined,
        "zip": undefined,
      },
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "LongText",
    "questionText": "Service Address",
  },
  {
    "assessmentNumber": "CUS02",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "LongText",
    "questionText": "Associated Providers",
  },
  {
    "assessmentNumber": "CUS04",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "LongText",
    "questionText": "Case Manager",
  },
  {
    "assessmentNumber": "CUS06",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "LongText",
    "questionText": "Contacts",
  },
  {
    "assessmentNumber": "M0150",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectAllThatApply",
    "questionText": "(C1) (M0150) CURRENT PAYMENT SOURCES FOR HOME CARE: (MARK ALL THAT APPLY.)",
  },
  {
    "assessmentNumber": "M0063",
    "checkedResponse": {
      "choice": "NA",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Medicare Number",
  },
  {
    "assessmentNumber": "M0065",
    "checkedResponse": {
      "choice": "NA",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Medicaid Number",
  },
  {
    "assessmentNumber": "M0150_specify",
    "checkedResponse": {
      "choice": "NA",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "INDICATE OTHER FINANCIAL FACTOR(S)",
  },
  {
    "assessmentNumber": "CUS01",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Payor Sources",
  },
  {
    "assessmentNumber": "CUS05",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Patient Phone Number",
  },
  {
    "assessmentNumber": "CUS024",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Patient Mobile Phone Number",
  },
  {
    "assessmentNumber": "CUS025",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Patient Alternate Phone Number",
  },
  {
    "assessmentNumber": "CUS03",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "LongText",
    "questionText": "Patient Referral Source",
  },
  {
    "assessmentNumber": "M0030",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "Date",
    "questionText": "SOC Visit Date",
  },
  {
    "assessmentNumber": "M0090",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "Date",
    "questionText": "Date Assessment Completed",
  },
  {
    "assessmentNumber": "AC1",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "SOC Visit Start Time",
  },
  {
    "assessmentNumber": "CUS08",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Minutes Spent with Patient",
  },
  {
    "assessmentNumber": "M0104",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "Date",
    "questionText": "(M0104) DATE OF REFERRAL: INDICATE THE DATE THAT THE WRITTEN OR VERBAL REFERRAL FOR INITIATION OR RESUMPTION OF CARE WAS RECEIVED BY THE HHA.",
  },
  {
    "assessmentNumber": "M0102_1",
    "checkedResponse": {
      "choice": "1. YES",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "(E)(M0102) IS THE DATE OF PHYSICIAN-ORDERED START OF CARE / RESUMPTION OF CARE KNOWN?",
  },
  {
    "assessmentNumber": "M0102_2",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "Date",
    "questionText": "(M0102) DATE OF PHYSICIAN-ORDERED START OF CARE (RESUMPTION OF CARE): IF THE PHYSICIAN INDICATED A SPECIFIC START OF CARE (RESUMPTION OF CARE) DATE WHEN THE PATIENT WAS REFERRED FOR HOME HEALTH SERVICES, RECORD THE DATE SPECIFIED.",
  },
  {
    "assessmentNumber": "CUS20",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Drive Start Time",
  },
  {
    "assessmentNumber": "CUS21",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Drive End Time",
  },
  {
    "assessmentNumber": "CUS22",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "SOC Visit End Time",
  },
  {
    "assessmentNumber": "CUS23",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Vital Sign Measurement time",
  },
  {
    "assessmentNumber": "AC1040_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on height exception",
  },
  {
    "assessmentNumber": "AC1045_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on weight exception",
  },
  {
    "assessmentNumber": "AC1050_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on temperature exception",
  },
  {
    "assessmentNumber": "AC1060_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on pulse exception",
  },
  {
    "assessmentNumber": "AC1070_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on respirations exception",
  },
  {
    "assessmentNumber": "AC1080_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on systolic BP exception",
  },
  {
    "assessmentNumber": "AC1083_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on diastolic BP exception",
  },
  {
    "assessmentNumber": "AC1091_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on oxygen saturation exception",
  },
  {
    "assessmentNumber": "AC1101_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on pain level exception",
  },
  {
    "assessmentNumber": "AC1111_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on blood sugar exception",
  },
  {
    "assessmentNumber": "AC1121_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on INR level exception",
  },
  {
    "assessmentNumber": "AC1131_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on prothrombin time exception",
  },
  {
    "assessmentNumber": "AC1141_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on ankle circumference exception",
  },
  {
    "assessmentNumber": "AC1151_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on calf circumference exception",
  },
  {
    "assessmentNumber": "AC1161_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on girth exception",
  },
  {
    "assessmentNumber": "AC1171_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on head circumference exception",
  },
  {
    "assessmentNumber": "AC1181_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on instep circumference exception",
  },
  {
    "assessmentNumber": "AC1191_exception_comment",
    "checkedResponse": {
      "choice": "test exception notes",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Comments on thigh circumference exception",
  },
  {
    "assessmentNumber": "CUS12",
    "checkedResponse": {
      "choice": "YES",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "Was the patient's physician notified about medication interactions?",
  },
  {
    "assessmentNumber": "CUS18",
    "checkedResponse": {
      "choice": "YES",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "Was the patient's physician notified about vital sign exceptions?",
  },
  {
    "assessmentNumber": "CUS13",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "BIMS Summary Score",
  },
  {
    "assessmentNumber": "D0160",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Total Severity Score:",
  },
  {
    "assessmentNumber": "CUS16",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "ACCORDING TO THE MAHC 10 FALL RISK ASSESSMENT, THIS PATIENT'S SCORE IS:",
  },
  {
    "assessmentNumber": "CUS17",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "BASED ON THE SCORE, THE PATIENT IS:",
  },
  {
    "assessmentNumber": "CUS14",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "TOTAL NUTRITION ASSESSMENT SCORE:",
  },
  {
    "assessmentNumber": "CUS15",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "BASED ON THE SCORE, THE NUTRITIONAL RISK LEVEL IS:",
  },
  {
    "assessmentNumber": "PA-fa9e30",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeFormNumber",
    "questionText": "TOTAL SCORE (PATIENTS WITH A TOTAL SCORE OF 12 OR LESS ARE CONSIDERED TO BE AT HIGH RISK OF DEVELOPING PRESSURE ULCERS):",
  },
  {
    "assessmentNumber": "PA-f83cf9",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "INDICATE THE BRADEN RISK LEVEL PRESENTED:",
  },
  {
    "assessmentNumber": "CUS19",
    "checkedResponse": {
      "choice": "No height data available",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Patient Height",
  },
  {
    "assessmentNumber": "PA-ec76c0",
    "checkedResponse": {
      "choice": "YES - CONTINUE TO PHQ9 INTERVIEW",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "SelectOne",
    "questionText": "DID THE PATIENT RESPOND WITH A SYMPTOM PRESENCE OF "NOT ASSESSED/NO INFORMATION" OR A SYMPTOM FREQUENCY OF "7 - 11 DAYS" OR "12 - 14 DAYS"?",
  },
  {
    "assessmentNumber": "AC1045-1",
    "checkedResponse": {
      "choice": null,
    },
    "integrationMessage": null,
    "integrationStatus": "Completed",
    "mappedType": "FreeFormNumber",
    "questionText": "What is the patient's measured weight, in kilogram?",
  },
  {
    "assessmentNumber": "M0080",
    "checkedResponse": {
      "choice": "1-RN",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Discipline of Person Completing Assignment",
  },
  {
    "assessmentNumber": "M0100",
    "checkedResponse": {
      "choice": "1 - Start of Care - FURTHER VISITS PLANNED",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "(OBQI)(M0100) THIS ASSESSMENT IS CURRENTLY BEING COMPLETED FOR THE FOLLOWING REASON:",
  },
  {
    "assessmentNumber": "M0110",
    "checkedResponse": {
      "choice": "1 - Early",
    },
    "integrationMessage": null,
    "integrationStatus": "Pending",
    "mappedType": "FreeForm",
    "questionText": "Episode Timing",
  },
]
`;
